reference/binary_model/extending.rst

```diff
@@ -0,0 +1,155 @@
+  (just a typically faster build thanks to Ninja).
+
+There are some exceptions to the above. For example, settings can be defined per-package using the ``<pattern:>setting=value``, both in profiles and
+command line:
+
+.. code-block:: bash
+
+    $ conan install . -s mypkg/*:compiler=gcc -s compiler=clang ..
+
+This will use ``gcc`` for "mypkg" and ``clang`` for the rest of the dependencies (in most cases it is recommended to use the same compiler for the whole dependency graph, but some scenarios when strong binary compatibility is guaranteed, it is possible to mix libraries built with different compilers).
+
+There are situations whereby many packages use the same option value, thereby allowing you to set its value once using patterns, like:
+
+.. code-block:: bash
+
+    $ conan install . -o *:shared=True
+
+
+Custom configuration
+--------------------
+
+As commented above, the Conan ``conf`` configuration system is intended to tune some of the tools and behaviors, but without really affecting the resulting package binaries. Some typical ``conf`` items are activating parallel builds, configuring "retries" when uploading to servers, or changing the CMake generator.
+Read more about :ref:`the Conan configuration system in this section<reference_config_files_global_conf>`.
+
+There is also the possibility to define ``user.xxxx:conf=value`` for user-defined configuration, that in the same spirit as core and tools built-in configurations, do not affect the ``package_id`` of binaries.
+
+But there might be some special situations in which is really desired that some ``conf`` defines different ``package_ids``, creating different package binaries. It is possible to do this in two different places:
+
+- Locally, in recipeps ``package_id`` method, via the ``self.info.conf`` attribute:
+
+  .. code-block:: python
+
+        def package_id(self):
+            # We can get the value from the actual current conf value, or define a new value
+            value = self.conf.get("user.myconf:myitem")
+            # This ``self.info.conf`` will become part of the ``package_id``
+            self.info.conf.define("user.myconf:myitem", value)
+
+- Globally, with the ``tools.info.package_id:confs`` configuration, receiving as argument a list of existing configuration to be part of the package ID, so you can define in profiles:
+
+  .. code-block:: ini
+
+    tools.info.package_id:confs=["tools.build:cxxflags", ...]
+
+  The value of the ``package_id`` will contain the value provided in the ``tools.build:cxxflags`` and other configurations. Note that this value is managed as a string, changing the string, will produce a different result and a different ``package_id``, so if this approach is used, it is very important to be very consistent with the provided values for different configurations like ``tools.build:cxxflags``.
```

Comment:
Should it maybe mention that it can also use * if you are sure enough that you want all your `user.myteam.*` confs to affect the pkg id?

Reply:
Sure, adding it.